src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -488,17 +490,97 @@ G1RemSet::G1RemSet(G1CollectedHeap* g1h,
   _g1h(g1h),
   _ct(ct),
   _g1p(_g1h->policy()),
-  _hot_card_cache(hot_card_cache) {
+  _hot_card_cache(hot_card_cache),
+  _sampling_task(NULL) {
 }
 
 G1RemSet::~G1RemSet() {
   delete _scan_state;
+  if (_sampling_task != NULL) {
```

Comment:
Could you add a comment to explain when `_sampling_task` would be null?

Reply:
Actually the check can be removed because the operator implementation handles NULL (by ignoring it).

Reply:
Even better. Thanks for the suggestion.

Reply:
Cool, then I will just remove the NULL-check.